server/src/domain/media/media.util.ts

```diff
@@ -521,9 +531,19 @@ export class QSVConfig extends BaseHWConfig {
 export class VAAPIConfig extends BaseHWConfig {
   getBaseInputOptions() {
     if (this.devices.length === 0) {
-      throw Error('No VAAPI device found');
+      throw new Error('No VAAPI device found');
     }
-    return [`-init_hw_device vaapi=accel:/dev/dri/${this.devices[0]}`, '-filter_hw_device accel'];
+
+    let hwDevice = `/dev/dri/${this.devices[0]}`;
+
+    if (this.config.preferredHwDevice !== 'auto') {
+      if (!this.devices.includes(this.config.preferredHwDevice.replace('/dev/dri/', ''))) {
+        throw new Error(`Device '${this.config.preferredHwDevice}' does not exist`);
+      }
+      hwDevice = this.config.preferredHwDevice;
+    }
```

Comment:
This is duplicated in the QSVConfig. Do you think it makes sense to move this part to some utility method in BaseHWConfig and call it in both places?

Reply:
I Will do It tomorrow, a function that retorns either a string with the node or null